### PR TITLE
Introduced TriaHelper class

### PR DESCRIPTION
**Pull request by asartori86**
In pidomus we had `shared_ptr<parallel::distributed::Triangulation<dim,spacedim> > triangulation` which was used for both the serial and parallel jobs. In order to have a proper triangulation, i.e. serial when we choose `LADealII` and a `p::d::` one otherwise, I introduced an helper class named TriaHelper.

I would like to delegate to this class also the operation about mesh refinement. But in a follow up PR to keep changes small

### CODE REVIEW

**Comment by asartori86** [comment]
the issue was that we used a serial mesh refinement providing a `parallel::distributed:triangulation`. Now that we fixed the tria management, all and only the output of the tests were mesh refinement was involved needed to be updated.

File: include/lac/lac_type.h [hunk 1]
```diff
@@ -36,6 +36,7 @@ class LADealII
   typedef BlockVector<double> VectorType;
   typedef BlockSparseMatrix<double> BlockMatrix;
   typedef dealii::BlockSparsityPattern BlockSparsityPattern;
+  static const bool TriaType = true; // serial
```
Comment (luca-heltai) [inline include/lac/lac_type.h]: This cannot be called "TriaType", if its meaning is "triangulation_is_serial".

Comment (asartori86) [reply include/lac/lac_type.h]: I agree. rename to "TriaSerial" ? other suggestions? Should I do an enum?

**Comment by asartori86** [comment]
@luca-heltai do you like this way?

**Comment by asartori86** [comment]
@luca-heltai ping

**Pull request merged**